### PR TITLE
Fix transaction submission for hex

### LINKED ISSUE #249: author_submitExtrinsic extrinsic encoding be hex data
For Poc-1, calling the `author_submitExtrinsic` RPC with `"params": ["0xfff745..."]` works as expected. 

Currently on master, the serialisation expects an array of bytes, i.e. the above needs to be encoded as `[255, 245, 69, ...]` This doesn't align with any of the other RPC requests or results where "blobs of data" are always expected/returned as hex-encoded strings.

If a "rich" encoding is desirable as an RPC, then it should be done with a second RPC.

### PR DESCRIPTION
**Pull request by gavofyork**
Closes #249

### CODE REVIEW

File: substrate/rpc/src/author/error.rs [hunk 1]
```diff
@@ -24,6 +24,11 @@ error_chain! {
 		Pool(txpool::Error, txpool::ErrorKind) #[doc = "Pool error"];
 	}
 	errors {
+		/// Not implemented yet
```
Comment (pepyakin) [inline substrate/rpc/src/author/error.rs]: I guess the comment is wrong.

File: substrate/primitives/src/lib.rs [hunk 2]
```diff
@@ -87,3 +89,17 @@ pub type AuthorityId = [u8; 32];
 
 /// A 512-bit value interpreted as a signature.
 pub type Signature = hash::H512;
+
+/// Contract storage key.
```
Comment (pepyakin) [inline substrate/primitives/src/lib.rs]: Why this says "contract storage key"?

File: substrate/rpc/src/author/mod.rs [hunk 3]
```diff
@@ -35,8 +37,11 @@ build_rpc_trait! {
 	/// Substrate authoring RPC API
 	pub trait AuthorApi<Hash, Extrinsic> {
 		/// Submit extrinsic for inclusion in block.
+		#[rpc(name = "author_submitRichExtrinsic")]
+		fn submit_rich_extrinsic(&self, Extrinsic) -> Result<Hash>;
```
Comment (pepyakin) [inline substrate/rpc/src/author/mod.rs]: I'm not sure about naming scheme choosen here. I kinda understand why it is "rich" extrinsic, but maybe it is better to leave it as "submit_extrinsic" and make another procedure call "submit_raw_extrinsic"? 

Or are we constrained by backward compatibility here?

Comment (rphmeier) [reply substrate/rpc/src/author/mod.rs]: most practical uses of substrate will be easier to use with the raw variant, so it makes sense to me to optimize the API a bit for it.

**Comment by jacogr** [comment]
Just one comment -just checking if it is as-intended. There is a mismatch here with regards to the previous iteration. Previously could submit just the data hex, i.e `0xff7769...`, now it would seem it needs `0x<length>ff7769...`

**Comment by gavofyork** [comment]
that's intended - it's the full form of the transaction.

**Pull request merged**